docs/man/ctags-faq.7.rst

```diff
@@ -0,0 +1,401 @@
+
+1.  Within each of the leaf nodes (i.e. ``hdrs``, ``lib``, ``src``, ``test``) build a tag
+    file using "``ctags *.[ch]``". This can be easily be done for the whole
+    hierarchy by making a shell script, call it ``dirtags``, containing the
+    following lines:
+
+	.. code-block:: sh
+
+		#!/bin/sh
+		cd $1
+		ctags *
+
+    Now execute the following command:
+
+	.. code-block:: sh
+
+		find * -type d -exec dirtags {} \;
+
+    These tag files are trivial (and extremely quick) to rebuild while
+    making changes within a directory. The following Vim key mapping is
+    quite useful to rebuild the tag file in the directory of the current
+    source file:
+
+	.. code-block:: text
+
+		:nmap ,t :!(cd %:p:h;ctags *.[ch])&<CR><CR>
+
+2.  Build the global tag file:
+
+	.. code-block:: sh
+
+		cd ~/project
+		ctags --file-scope=no -R
+
+    thus constructing a tag file containing only non-static tags for all
+    source files in all descendent directories.
+
+3.  Configure your editor to read the local tag file first, then consult
+    the global tag file when not found in the local tag file. In Vim,
+    this is done as follows:
+
+	.. code-block:: text
+
+		:set tags=./tags,tags,~/project/tags
+
+If you wish to implement Approach 4, you would need to replace the
+``dirtags`` script of step 1 with the following:
+
+	.. code-block:: sh
+
+		#!/bin/sh
+		cd $1
+		ctags *
+		# Now append the non-static tags from descendent directories
+		find * -type d -prune -print | ctags -aR --file-scope=no -L-
+
+And replace the configuration of step 3 with this:
+
+	.. code-block:: text
+
+		:set tags=./tags;$HOME,tags
+
+As a caveat, it should be noted that step 2 builds a global tag file whose
+file names will be relative to the directory in which the global tag file
+is being built. This takes advantage of the Vim ``tagrelative`` option,
+which causes the path to be interpreted a relative to the location of the
+tag file instead of the current directory. For standard vi, which always
+interprets the paths as relative to the current directory, we need to
+build the global tag file with absolute path names. This can be
+accomplished by replacing step 2 with the following:
+
+	.. code-block:: sh
+
+		cd ~/project
+		ctags --file-scope=no -R `pwd`
+
+Does Universal Ctags support Unicode file names?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. MEMO: from https://github.com/universal-ctags/ctags/issues/1837
+
+Yes, Unicode file names are supported on unix-like platforms (Linux, macOS,
+Cygwin, etc.).
+
+However, on Windows, you need to use Windows 10 version 1903 or later to use
+Unicode file names. (This is an experimental feature, though.) On older versions
+on Windows, Universal Ctags only support file names represented in the current
+code page. If you still want to use Unicode file names on them, use Cygwin or
+MSYS2 version of Universal Ctags as a workaround.
+
+Why does zsh cause "zsh: no matches found" error?
```

Comment:
I found tcsh made the similar warning, too.
```
[jet@living]~/var/ctags-new% tcsh
[jet@living ctags-new]$ echo --foo=fnmatch/*
echo: No match.
[jet@living ctags-new]$ ls --*
ls: No match.
```

Reply:
Interesting!  This may be the root cause of the "feature" of zsh.

```
echo: No match.
ls: No match.
```

Filename expansion is a job of a shell.  (On MS-DOS it is a job of each command.)
Either `echo` nor `ls` caused the error. "tcsh: No match" like on zsh is a correct error message.

I thought about adding a comment that tcsh causes similar error, but I did not.
It is because I think we should not take care of csh.  cf. [Csh Programming Considered Harmful](http://www.faqs.org/faqs/unix-faq/shell/csh-whynot/)